agent-inject/agent/annotations.go

```diff
@@ -553,6 +559,10 @@ func Init(pod *corev1.Pod, cfg AgentConfig) error {
 		pod.ObjectMeta.Annotations[AnnotationTemplateConfigMaxConnectionsPerHost] = strconv.FormatInt(cfg.MaxConnectionsPerHost, 10)
 	}
 
+	if _, ok := pod.ObjectMeta.Annotations[AnnotationTemplateConfigLeaseRenewalThreshold]; !ok {
```

Comment:
It might be good to add tests for this new annotation.

Reply:
Added [here](https://github.com/hashicorp/vault-k8s/pull/761/files#diff-8be5beafe59d3cada9d7b742473b1b0fd003e2b4d661a167e63da1d8d121e594R1483)